posthog/models/entity.py

```diff
@@ -69,6 +69,19 @@ def equals(self, other) -> bool:
 
         return True
 
+    """ Checks if this entity is a generalized version of other"""
+
+    def includes(self, other) -> bool:
+        self_properties = sorted([str(prop) for prop in self.properties])
+        other_properties = sorted([str(prop) for prop in other.properties])
+
+        num_matched_props = sum([1 for x, y in zip(self_properties, other_properties) if x == y])
+
+        if self.id == other.id and num_matched_props == len(self_properties):
```

Comment:
Proposal:

```python
return self.id == other.id and all(str(self_prop) == str(other_prop) for self_prop, other_prop in zip(self.properties, other.properties)
```

This simplifies this code significantly and is more idiomatic python

Reply:
Note: I might have misunderstood this, a length check might be needed as well (len(self.properties) >= len(other.properties

Reply:
i'm changing it back. I forgot what i was checking at first, but we're not checking for equality here. We're just checking that the calling entity is a subset of the called on entity